src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx

```diff
@@ -137,6 +139,15 @@ export const BorrowInput = ({
     if (!borrowAsset) setBorrowAsset(collateralAsset)
   }, [borrowAsset, borrowAssets, collateralAsset, setBorrowAsset])
 
+  // If the user disconnects the chain for the currently selected borrow asset, default to the collateral asset
+  useEffect(() => {
+    if (!collateralAsset || !borrowAsset) return
+
+    if (!walletConnectedChainIds.includes(borrowAsset.chainId)) {
+      setBorrowAsset(collateralAsset)
+    }
+  }, [collateralAsset, borrowAsset, setBorrowAsset, walletConnectedChainIds])
+
```

Comment:
And same goes for borrows, see https://github.com/shapeshift/web/pull/7029#discussion_r1622042222

A user may have disconnected both their collateral asset *and* their currently selected borrow asset

https://github.com/shapeshift/web/assets/17035424/bd3814f2-e2ba-4e4c-9b57-4208258a750c



tl;dr: while being smort re: borrow/repayment asset wallet support reactivity in borrows and repays instead of making wallet-support assumptions, we make another assumption that the collateral asset is still supported.

Reply:
Yeah whatever we land on, it should balance user benefit with cost and complexity
https://github.com/shapeshift/web/pull/7029#discussion_r1623669765